### PR TITLE
[core][iOS] Allow export of views that conform to `AnyExpoView`

### PR DESCRIPTION
**Pull request by dominicstop**
hello! (big thanks to the expo team + contributors for all your hard work in making an outstanding platform; this is my first PR so i apologize for the mistakes)

# Why

i want to export a view that inherits from `RCTView`, while still being able to get the `AppContext`. This is useful for exporting views that subclass existing react views (e.g. `RCTBaseTextInputView`, etc).
<br>

# How

* Created a protocol called [`AnyExpoView`](https://github.com/dominicstop/expo-fork-for-pr-01/blob/main/packages/expo-modules-core/ios/Core/Protocols/AnyExpoView.swift).
* Updated [`ExpoView`](https://github.com/dominicstop/expo-fork-for-pr-01/blob/main/packages/expo-modules-core/ios/Core/Views/ExpoView.swift) to conform to `AnyExpoView`.
* Updated [`ViewDefinition.createView(appContext:)`](https://github.com/dominicstop/expo-fork-for-pr-01/blob/b4a31860d7d83ee5145102305f4503e26d6b6686/packages/expo-modules-core/ios/Core/Views/ViewDefinition.swift#L43-L46) function to initialize + return `AnyExpoView`.

<br>

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by tsapeta — CHANGES_REQUESTED** [review]
That looks great, thank you! Congrats on your first contribution 👏 I left just one comment to remove the header comment and unnecessary `UIKit` import (it's not needed here and would break on macOS where it's unavailable).

What a coincidence because we just had a similar discussion with @nandorojo about subclassing views other than just `RCTView` 😉

File: packages/expo-modules-core/ios/Core/Protocols/AnyExpoView.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//
+//  AnyExpoView.swift
+//  ExpoModulesCore
+//
+//  Created by Dominic Go on 2/26/24.
+//
+import UIKit
```
[the remark this answers is not in the document]
Comment (dominicstop) [reply packages/expo-modules-core/ios/Core/Protocols/AnyExpoView.swift]: hello, thank you for responding; i have made the changes you requested (i didn't realize you could apply the changes via the web UI, sorry)

**Review by tsapeta — APPROVED** [review]

File: packages/expo-modules-core/ios/Core/Protocols/AnyExpoView.swift [hunk 2]
```diff
@@ -0,0 +1,5 @@
+public protocol AnyExpoView: RCTView {
+  var appContext: AppContext? { get }
+
+  init(appContext: AppContext?)
+}
```
Comment (lukmccall) [inline packages/expo-modules-core/ios/Core/Protocols/AnyExpoView.swift:5]: I'm wondering if there is a need for similar features on Android 🤔

**Review by lukmccall — APPROVED** [review]
Looks good

**Pull request merged**